### PR TITLE
DAP terraform

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1710)
We need to capture our current DAP infrastructure in Terraform. 

## This PR
* Creates a "base" environment at `environments/dap`, with concrete instantations of that environment in subdirs `dev/` and `prod/`. The intent is to avoid our current pattern of copying+pasting the world around. The DAP base is also not generic enough to warrant placement in our `templates` dir, hence this approach. Would love feedback on this approach.

### CODE REVIEW

**Review by quazi-broad — APPROVED** [review]
lgtm. I'll refer to this if I get to pick up that HCA TF ticket!

**Review by raaidbroad — APPROVED** [review]
I like this. It's simple, clean, easy to read; all about it

File: .github/workflows/terraformPlan.yml [hunk 1]
```diff
@@ -17,6 +17,14 @@ jobs:
...
       matrix:
         include:
+        - environment: dap/dap-dev
+          vault_env: dev
+          vault_role_id_secret: MONSTER_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_AUTH_ROLE_SECRET
+        - environment: dap/dap-prod
+          vault_env: prod
+          vault_role_id_secret: MONSTER_PROD_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_PROD_AUTH_ROLE_SECRET
```
Comment (ThePletch) [inline .github/workflows/terraformPlan.yml:27]: really wish matrices supported nested hashes or templates or something, this repetition is so tedious

File: environments/dap/base/buckets.tf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+resource google_storage_bucket storage_bucket {
```
Comment (ThePletch) [inline environments/dap/base/buckets.tf:1]: i'm finding myself questioning what we get out of dividing out resources so granularly in cases like this where there's just one resource. is consistency worth the increased number of files to reference?

File: environments/dap/base/variables.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+variable project_name {
+  type = string
+}
+
+variable vault_prefix {
+  type = string
```
Comment (ThePletch) [inline environments/dap/base/variables.tf:6]: can we include descriptions for variables like this? i'd want to know things like "what the prefix should start with" and "what will be appended to it" for a `vault_prefix` variable

File: templates/terraform/api-services/variables.tf [hunk 4]
```diff
@@ -2,3 +2,28 @@ variable service_ids {
...
+    "runtimeconfig.googleapis.com",
+    "sqladmin.googleapis.com",
+    "sql-component.googleapis.com",
+    "stackdriver.googleapis.com",
+    "storage-api.googleapis.com",
+    "storage-component.googleapis.com",
+    "dataflow.googleapis.com",
+    "storagetransfer.googleapis.com",
+    "pubsub.googleapis.com"
+  ]
```
Comment (ThePletch) [inline templates/terraform/api-services/variables.tf:28]: anywhere else we can use this to clean up existing modules?

File: environments/dap/base/buckets.tf (hunk 2 above)
Comment (ThePletch) [reply environments/dap/base/buckets.tf:1]: like, to me, the benefit of file splits is knowing exactly where to find a resource in a large module, but for small modules like this, it seems like it'd be easier to just keep the config in a single file.

File: environments/dap/dap-prod/terraform/dap.tf [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module dap {
+  source       = "../../base"
+  project_name = "broad-dsp-monster-dap-prod"
+  vault_prefix = "secret/dsde/monster/dev/dog-aging/redcap-tokens/automation"
```
Comment (ThePletch) [inline environments/dap/dap-prod/terraform/dap.tf:4]: this is another point in favor of `vault_prefix` having a description - i didn't realize it was the path for the redcap token.

it also looks like the value is currently unused?

**Review by ThePletch — COMMENTED** [review]
i strongly prefer this approach of a central app-specific module with a configured instance in each environment, but in this particular case it seems like there's a bit of clutter to eliminate.

File: environments/dap/base/buckets.tf (hunk 2 above)
Comment (aherbst-broad) [reply environments/dap/base/buckets.tf:1]: Good call. I'll merge this into a central "resources.tf" and we can refactor from there.

File: environments/dap/dap-prod/terraform/dap.tf (hunk 5 above)
Comment (aherbst-broad) [reply environments/dap/dap-prod/terraform/dap.tf:4]: 🤔 yeah this seems redundant, will remove.

File: templates/terraform/api-services/variables.tf (hunk 4 above)
Comment (aherbst-broad) [reply templates/terraform/api-services/variables.tf:28]: https://broadinstitute.atlassian.net/browse/DSPDC-1718

**Pull request merged**